font/src/lib.rs

```diff
@@ -135,39 +140,48 @@ impl FontKey {
     }
 }
 
-#[derive(Debug, Copy, Clone, Eq)]
-pub struct GlyphKey {
-    pub c: char,
-    pub font_key: FontKey,
-    pub size: Size,
+/// Captures possible outcomes of shaping, if shaping succeeded it will return a `GlyphIndex`.
+/// If shaping failed or did not occur, `Fallback` will be returned.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum KeyType {
+    /// Shaping returned a valid index and we can render that as expected.
+    GlyphIndex(u32),
+    /// Shaping returned a missing glyph or shaping did not occur. If glyph is missing system will
+    /// attempt to load character glyph from a fallback font. If shaping did not occur this will
+    /// first try the configured font then fallback.
+    Fallback(char),
+    /// Placeholder glyph useful when we need a glyph but it shouldn't ever render as anything
+    /// (cursors, wide_char_spacers, etc.)
+    Placeholder,
 }
 
-impl Hash for GlyphKey {
-    fn hash<H: Hasher>(&self, state: &mut H) {
-        unsafe {
-            // This transmute is fine:
-            //
-            // - If GlyphKey ever becomes a different size, this will fail to compile
-            // - Result is being used for hashing and has no fields (it's a u64)
-            ::std::mem::transmute::<GlyphKey, u64>(*self)
-        }
-        .hash(state);
+impl Default for KeyType {
+    // Arbitrary non missing (non 0) glyph
+    // Used as a placeholder when glyph doesn't matter, such as when representing a cursor.
```

Comment:
I think with having this `PLACEHOLDER_GLYPH` as constant, the `non 0` comment is more confusing than helpful. It also doesn't seem arbitrary now, since it's a well named constant. So I'd just remove this comment.

Since this is the font library, we also shouldn't conflate it with topics like 'cursors', that's a terminal and not a font thing. As a library it's up to the user what they want to use this for.

So in general I think the useful part of this comment is that it hints at the default glyph key being completely empty, but I don't think that's super crucial information and pretty much the only thing that makes sense.

So I'd remove both of these comments myself, though if you think they bring some good value I'd recommend shortening them down to just a small statement saying that the default glyph key is for an empty glyph.